.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,77 @@
+name: Build, test and push a Docker image
+
+on:
+  push:
+    branches:
+      - main
+      - gcp-stage  # FIXME(smarnach): Remove after the GCP migration
```

Comment:
I don't understand how this branch is going to be used. In the PR description is this line:

> The workflow also builds a stage image when we push to the gcp-stage branch, to make it easier to build test images. We can remove this after the GCP migration.

But I don't understand what that entails and I don't see any explanation of how we're structuring the interim period when we're deploying to both AWS and GCP environments in either the Tecken GCP project plan or OBS-268 issue.

Can you write up the plan somewhere so that we have a reference to review against?

Reply:
Sorry, I should have explained this better. This branch isn't really supposed to be part of the migration plan. I mainly planned to use it for two things:

* Build images for GCP before we have all the storage interface PRs merged. Currently, `main` is simply not suitable for that, and it will take at least another few days until it is. Since I'm already testing things in GCP, I wanted a convenient way of creating images for that. There is no automatic deployment yet – it's just so I don't have to build the test images manually.

* Build test images when working on the system tests. We have a few tickets to expand the system tests. It's likely they will fail against the new environment at some point, and then I'll need to prepare a fix. The only way to test that the fix actually works is deploying it to the GCP test environment, since the dev environment simply isn't similar enough in the aspects we are testing now, like response headers and HEAD requests.

Outside of these two cases, I don't intent to deviate from our usual model – `main` will be deployed to stage and version tags will be deployed to prod in both AWS and GCP.

I can also build images manually for these two cases. It's not a big issue. It just seemed more convenient and less error-prone to be able to force-push to `gcp-stage` to get a fresh image during the early GCP test period.

And I know I need to fill in more details in the migration plan; I'll do that later this week.

Reply:
I changed this a bit. First, I renamed the branch to `build-test-image`, because that's the intention. Second, the image is tagged `test` rather than `latest`. And third, the image is not automatically deployed anywhere.